api/repositories/role_repository_test.go

```diff
@@ -0,0 +1,218 @@
+package repositories_test
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"code.cloudfoundry.org/cf-k8s-controllers/api/repositories"
+	"code.cloudfoundry.org/cf-k8s-controllers/api/repositories/fake"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	hnsv1alpha2 "sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+var _ = Describe("RoleRepository", func() {
+	var (
+		ctx                 context.Context
+		rootNamespace       string
+		roleRepo            *repositories.RoleRepo
+		roleRecord          repositories.RoleRecord
+		authorizedInChecker *fake.AuthorizedInChecker
+	)
+
+	BeforeEach(func() {
+		rootNamespace = uuid.NewString()
+		ctx = context.Background()
+		authorizedInChecker = new(fake.AuthorizedInChecker)
+		Expect(k8sClient.Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rootNamespace}})).To(Succeed())
+		roleRepo = repositories.NewRoleRepo(k8sClient, authorizedInChecker, map[string]string{"space_developer": "cf-space-dev-role"})
+
+		roleRecord = repositories.RoleRecord{}
+	})
+
+	Describe("CreateSpaceRole", func() {
+		createSubnamespaceAnchor := func(name, parent string) *hnsv1alpha2.SubnamespaceAnchor {
```

Comment:
Given we set SpaceNameLabel, this is creating a `space` subnamespaceanchor. Consider renaming.

Reply:
There is also some duplication with org/space creation in the org_repository_test code. Potential refactor.